generator/webapp/src/main/java/io/fabric8/maven/generator/webapp/WebAppGenerator.java

```diff
@@ -68,7 +73,30 @@ public WebAppGenerator(GeneratorContext context) {
     @Override
     public boolean isApplicable(List<ImageConfiguration> configs) {
         return shouldAddImageConfiguration(configs) &&
-               MavenUtil.hasPlugin(getProject(), "org.apache.maven.plugins:maven-war-plugin");
+               MavenUtil.hasPlugin(getProject(), "org.apache.maven.plugins:maven-war-plugin") && !isWarUsedWithFatJar(configs);
+    }
+
+    /**
+     * If we are using spring boot or wildfly swarm with a fat jar then we should not use the web app generator
+     * and should stick with the fat jar generators instead
+     */
+    protected boolean isWarUsedWithFatJar(List<ImageConfiguration> configs) {
+        MavenProject project = getContext().getProject();
+        String finalName = project.getBuild().getFinalName();
+        String artifactId = project.getArtifactId();
+        if (!Objects.equal(artifactId, finalName)) {
+            String plugin = null;
+            if (MavenUtil.hasPlugin(project, SPRING_BOOT_MAVEN_PLUGIN_GA)) {
```

Comment:
Awesome. I'm more than happy to move the 'disable webapp generator' logic into spring boot / wildfly swarm plugins if that works; its a little cleaner then. Folks could then disable, say, the spring-boot generator and the webapp generator would kick back in etc. 

We should be sure to document why spring boot / wildfly swarm plugins have higher priority to avoid folks accidentally breaking this later.